src/components/views/AudioResourceSettingsView.tsx

```diff
@@ -0,0 +1,620 @@
+    this.state = {
+      form: initForm<ResourceSettingsForm>(
+        {
+          name: undefined,
+          orderNumber: undefined,
+          slug: undefined
+        },
+        resourceRules
+      ),
+
+      resourceId: "",
+      resourceData: ResourceUtils.getMaterialTableResourceData(props.resource),
+      loading: false,
+      dataChanged: false,
+      canPlay: false
+    };
+  }
+
+  /**
+   * Component did mount life cycle handler
+   */
+  public componentDidMount = async () => {
+    const { keycloak } = this.props;
+
+    keycloak?.token && this.updateComponentData();
+  }
+
+  /**
+   * Component did update  life cycle handler
+   *
+   * @param prevProps previous props
+   */
+  public componentDidUpdate = async (prevProps: Props) => {
+    const { keycloak, resource } = this.props;
+
+    if (prevProps.resource !== resource) {
+      keycloak?.token && this.updateComponentData();
+    }
+  }
+
+  /**
+   * Component render method
+   */
+  public render = () => {
+    const { classes } = this.props;
+    const { loading, dataChanged, form } = this.state;
+    const { isFormValid } = form;
+
+    if (loading) {
+      return;
+    }
+
+    return (
+      <Box>
+        <Button
+          className={ classes.saveButton }
+          color="primary"
+          variant="outlined"
+          disabled={ !isFormValid || !dataChanged }
+          onClick={ this.onSaveChanges }
+        >
+          { strings.save }
+        </Button>
+        { this.renderField("name", strings.commonSettingsTexts.name, "text") }
+        { this.renderUploaderAndPreview() }
+        <Box mt={ 3 } mb={ 3 }>
+          <Divider/>
+        </Box>
+        <AdminOnly>
+          { this.renderAdvancedSettings() }
+        </AdminOnly>
+      </Box>
+    );
+  }
+
+  /**
+   * Renders text field
+   *
+   * @param key to look for
+   * @param placeholder placeholder text
+   * @param type text field type
+   * @returns rendered text field
+   */
+  private renderField = (key: keyof ResourceSettingsForm, placeholder: string, type: string) => {
+    const { form } = this.state;
+    const { values, messages: { [key]: message } } = form;
+
+    return (
+      <TextField
+        fullWidth
+        multiline={ type === "textarea" }
+        rows={ type === "textarea" ? 8 : undefined }
+        type={ type }
+        error={ message && message.type === MessageType.ERROR }
+        helperText={ message && message.message }
+        value={ values[key] || "" }
+        onChange={ this.onHandleResourceTextChange(key) }
+        onBlur={ this.onHandleBlur(key) }
+        name={ key }
+        variant="outlined"
+        label={ placeholder }
+      />
+    );
+  };
+
+  /**
+   * Renders table that contains properties data
+   */
+  private renderPropertiesTable = () => {
+    const { resourceData } = this.state;
+
+    if (!resourceData?.properties) {
+      return null;
+    }
+
+    return (
+      <MaterialTable
+        key={ nanoid() }
```

Comment:
Is this by design since the use of nanoid will cause this component to be re-rendered every time?